afqinsight/nn/pt_models.py

```diff
@@ -0,0 +1,401 @@
+import math
+import torch
+from dipy.utils.optpkg import optional_package
+from dipy.utils.tripwire import TripWire
+
+torch_msg = (
+    "To use afqinsight's pytorch models, you need to have pytorch "
+    "installed. You can do this by installing afqinsight with `pip install "
+    "afqinsight[tf]`, or by separately installing pytorch with `pip install "
+    "pytorch`."
+)
+
+torch, has_torch, _ = optional_package("torch", trip_msg=torch_msg)
+if has_torch:
+    import torch.nn as nn
+else:
+    # Since all model building functions start with Input, we make Input the
+    # tripwire instance for cases where pytorch is not installed.
+    Input = TripWire(torch_msg)
+    print("test")
+
+class mlp4(nn.Module):
```

Comment:
Per PEP8, classes should be CapsCase and functions should be lower_case (with snake_case wherever relevant).